presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -419,21 +450,9 @@ else if (parameterType.isPrimitive()) {
                     argumentNativeContainerTypes.add(parameterType);
                     argumentTypes.add(parseTypeSignature(type.value(), literalParameters));
 
-                    boolean currentNullFlag = false;
-                    if (i + 1 < method.getParameterCount()) {
-                        annotations = method.getParameterAnnotations()[i + 1];
-                        if (Stream.of(annotations).anyMatch(IsNull.class::isInstance)) {
-                            Class<?> isNullType = method.getParameterTypes()[i + 1];
-
-                            checkArgument(Stream.of(annotations).filter(Parser::isPrestoAnnotation).allMatch(IsNull.class::isInstance), "Method [%s] has @IsNull parameter that has other annotations", method);
-                            checkArgument(isNullType == boolean.class, "Method [%s] has non-boolean parameter with @IsNull", method);
-                            checkArgument(!Primitives.isWrapperType(parameterType), "Method [%s] uses @IsNull following a parameter with boxed primitive type: %s", method, parameterType.getSimpleName());
-
-                            nullableArgument = true;
-                            currentNullFlag = true;
-                            // skip @IsNull on the next parameter
-                            i++;
-                        }
+                    if (currentNullFlag) {
+                        // skip @IsNull on the next parameter
```

Comment:
This wording is confusing, since it sounds like we're skipping an annotation. Rephrase:

```java
// skip @IsNull parameter